corehq/apps/users/views/__init__.py

```diff
@@ -1004,3 +1004,12 @@ def location_restriction_for_users(request, domain):
         project.location_restriction_for_users = json.loads(request.POST["restrict_users"])
     project.save()
     return HttpResponse()
+
+
+@require_POST
+@require_superuser
```

Comment:
@wspride and @sjain28 may be able to comment more conclusively, but I believe the workflow will require the user to enter web user credentials, so it can use those for auth for this request.